File: benchmarks/ackermann.c
#include <stdio.h>

static long ackermann(long m, long n) {
    if (m == 0) {
        return n+1;
    } else if (n == 0) {
        return ackermann(m-1, 1);
    } else {
        return ackermann(m-1, ackermann(m, n-1));
    }
}

int main(void) {
    printf("%ld\n", ackermann(3L, 14L));
    return 0;
}

File: benchmarks/ackermann.ml

let rec ackermann m n =
    if (m == 0) then
        n + 1
    else if (n == 0) then
        ackermann (m - 1) 1
    else
        ackermann (m - 1) (ackermann m (n - 1))
;;

Printf.printf "%d\n" (ackermann 3 14);;
